armi/settings/caseSettings.py

```diff
@@ -135,11 +144,33 @@ def __repr__(self):
             self.__class__.__name__, self.caseTitle, total, altered
         )
 
+    def _settingIsOkayToGrab(self, key):
+        """
+        A way to check if specific settings can be grabbed out of the case settings.
+
+        Could be updated with other specific instances as necessary.
```

Comment:
"ok to grab" is a bit ambiguous in this context. Maybe refer to some of the utils using :py:func:`armi.utils.blah` linkages or similar?

Maybe `_directAccessOfSettingAllowed` or something?

in general I'm not super happy about this special casing that has to be applied to all settings. It seems like it might be nicer if the settings system was more flexible in being able to allow special derived settings more universally. 

That might be nice to add here in the `Notes` section of the docstring for future work.

Reply:
I very much agree that I don't like this. I tried to avoid doing it for a long time, but ultimately this was the only way I could see to completely prevent other code from grabbing the defaults in things like `cs["burnSteps"]` or `cs["cycleLengths"]` when the cycle information is actually entered by the user in `cycles`. 

On the other hand though, it isn't all that often that things are pulled from the case settings, at least from the perspective of slowing down the overall calculation.

I tried to make it a bit more universal by pulling the key-check into the function, which is being done regardless, but to me it still feels like this doesn't really belong here.

I will change the name of the method and also add the note that you suggested for now.